### PR TITLE
replace mimir.enabled property with mc provider flavor as mimir is everywhere in capi

### PR DESCRIPTION
**Pull request by QuentinBisson**
Before adding a new alerting rule into this repository you should consider creating an SLO rules instead.
SLO helps you both increase the quality of your monitoring and reduce the alert noise.

* How to create a SLO rule: https://github.com/giantswarm/sloth-rules#how-to-create-a-slo
* Documentation: https://intranet.giantswarm.io/docs/monitoring/slo-alerting/

---
This PR removes the specific mimir test cases for capa as all capi installations run mimir anyway so we do not need this distinction and this will make things a lot cleaner

### Checklist

- [x] Update CHANGELOG.md
- [ ] Add [Unit tests](https://github.com/giantswarm/prometheus-rules/#testing)
- [ ] Follow [Alert structure](https://github.com/giantswarm/prometheus-rules/#how-alerts-are-structured)
- [ ] Consider [creating a dashboard](https://docs.giantswarm.io/getting-started/observability/grafana/custom-dashboards/) ([guidelines](https://intranet.giantswarm.io/docs/product/ux/guidelines/dashboards/)) (if it does not exist already) to help oncallers monitor the status of the issue.
- [ ] Request review from oncall area, as well as team (e.g: `oncall-kaas-cloud` GitHub group).

### CODE REVIEW

**Review by ubergesundheit — APPROVED** [review]
LGTM

**Review by fiunchinho — APPROVED** [review]

File: test/tests/providers/capi/capa/kaas/turtles/alerting-rules/capi-cluster.rules.test.yml
Comment (hervenicol) [inline test/tests/providers/capi/capa/kaas/turtles/alerting-rules/capi-cluster.rules.test.yml:1]: Where do all these tests come from? Are these new tests you just wrote?

File: test/conf/providers [hunk 1]
```diff
@@ -1,4 +1,4 @@
 vintage/aws
 capi/capz
 capi/capa
-capi/capa-mimir
+
```
[the remark this answers is not in the document]
Comment (QuentinBisson) [reply test/conf/providers:4]:
```suggestion
```

File: test/tests/providers/capi/capa/kaas/turtles/alerting-rules/capi-cluster.rules.test.yml
Comment (QuentinBisson) [reply test/tests/providers/capi/capa/kaas/turtles/alerting-rules/capi-cluster.rules.test.yml:1]: No :D I cpied the one that exists under `test/tests/providers/capi/capz` to also be under `test/tests/providers/capi/capa` to make sure I did not break anything and I moved the existing ones under `test/tests/providers/capi/capa-mimir` to both `test/tests/providers/capi/capa` and `test/tests/providers/capi/capz`  to increase the coverage

**Review by hervenicol — APPROVED** [review]
LGTM

**Pull request merged**